Semester.py breaks due to spelling correction in questions.

Someone corrected the spelling in the V2017 forms to say:

> Hva er ditt generelle inntrykk av kurset 

instead of 

> Hva er det generelle intrykk av kurset.

This is great, but as semester.py explicitly declares the latter as the general question, it breaks for V2017 forms. If you change semester.py to the corrected spelling, it no longer works for previous semesters forms. This is a problem if you are working on several semesters of forms simultaneously. 

The specific error is:
> python3 src/semester.py
> Traceback (most recent call last):
>   File "src/semester.py", line 50, in <module>
>     main("./data/"+sem)
>   File "src/semester.py", line 41, in main
>     semester_data[course_name] = get_course_data(semester_dir+"/outputs/stats/"+f)
>   File "src/semester.py", line 29, in get_course_data
>     course[general_question] = course["questions"][general_question]
> KeyError: 'Hva er ditt generelle intrykk av kurset?'
> Makefile:35: recipe for target 'json' failed
> make: *** [json] Error 1

--- a/Makefile
+++ b/Makefile
@@ -7,6 +7,15 @@ install-mac: # mac only :)
 	brew install phantomjs
 	brew install rename
 	brew install pandoc
+	# TODO: Add LaTeX installation
+	pip3 install -r requirements.txt
+
+install-linux:
+	apt install python3
+	apt install python3-pip
+	# TODO: Add phantomjs installation (apt doesn't work!)
+	apt install pandoc
+	# TODO: Add LaTeX installation
 	pip3 install -r requirements.txt
 
 usernames:
@@ -72,6 +81,7 @@ web-preview: web
 upload_raw:
 	@echo "Mount KURS folder to /Volumes/KURS (mac) or similar before running:"
 	python3 src/upload_reports.py -v --input ./data --output /Volumes/KURS/ --semester $(SEMESTER)
+	# python3 src/upload_reports.py -v --input ./data --output "Z:/KURS/" --semester $(SEMESTER)
 
 score:
 	python3 ./src/score.py $(SEMESTER)

--- a/resources/scales.json
+++ b/resources/scales.json
@@ -1,546 +1,661 @@
 {
-    "Hva er ditt generelle intrykk av kurset?": {
-        "order": [
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Greit",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "all": [
-            "",
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Greit",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "ignore": [
-            ""
-        ]
-    },
-    "Hva synes du om emnets nivå?": {
-        "order": [
-            "For vanskelig",
-            "Noe vanskelig",
-            "Passe",
-            "Noe lett",
-            "Lett"
-        ],
-        "all": [
-            "",
-            "For vanskelig",
-            "Noe vanskelig",
-            "Passe",
-            "Noe lett",
-            "Lett"
-        ],
-        "ignore": [
-            ""
-        ]
-    },
-    "Hva synes du om arbeidsmengden sett i forhold til antall studiepoeng?": {
-        "order": [
-            "For høy",
-            "Høy",
-            "Passe",
-            "Lav",
-            "For lav"
-        ],
-        "all": [
-            "",
-            "For høy",
-            "Høy",
-            "Passe",
-            "Lav",
-            "For lav"
-        ],
-        "ignore": [
-            ""
-        ]
-    },
-    "Kvaliteten på forelesningene?": {
-        "order": [
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "all": [
-            "Vet ikke",
-            "",
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "ignore": [
-            "Vet ikke",
-            ""
-        ]
-    },
-    "Forelesningenes relevans og nytteverdi i forhold til å formidle det du trenger å lære?": {
-        "order": [
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "all": [
-            "Vet ikke",
-            "",
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "ignore": [
-            "Vet ikke",
-            ""
-        ]
-    },
-    "Kvaliteten på gruppeundervisningen i sin helhet?": {
-        "order": [
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "all": [
-            "Vet ikke",
-            "",
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "ignore": [
-            "Vet ikke",
-            ""
-        ]
-    },
-    "I hvilken grad forbedrer ukesoppgavene og gruppeundervisningen din forståelse av pensum?": {
-        "order": [
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "all": [
-            "Vet ikke",
-            "",
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "ignore": [
-            "Vet ikke",
-            ""
-        ]
-    },
-    "Er pensum og læringsmål i emnet godt definert?": {
-        "order": [
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "all": [
-            "Vet ikke",
-            "",
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "ignore": [
-            "Vet ikke",
-            ""
-        ]
-    },
-    "Hvordan vurderer du kursmateriellet som benyttes (lærebok m.m.)?": {
-        "order": [
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "all": [
-            "Vet ikke",
-            "",
-            "Særdeles bra",
-            "Meget bra",
-            "Bra",
-            "Grei",
-            "Mindre bra",
-            "Lite bra"
-        ],
-        "ignore": [
-            "Vet ikke",
-            ""
-        ]
-    },
-    "Hvordan er nivået på obligene?": {
-        "order": [
-            "For høyt",
-            "Høyt",
-            "Passe",
-            "Lavt",
-            "For lavt"
-        ],
-        "all": [
-            "Ikke aktuelt",
-            "",
-            "For høyt",
-            "Høyt",
-            "Passe",
-            "Lavt",
-            "For lavt"
-        ],
-        "ignore": [
-            "Ikke aktuelt",
-            ""
-        ]
-    },
-    "Hva synes du om antallet obliger?": {
-        "order": [
-            "For høyt",
-            "Høyt",
-            "Passe",
-            "Lavt",
-            "For lavt"
-        ],
-        "all": [
-            "Ikke aktuelt",
-            "",
-            "For høyt",
-            "Høyt",
-            "Passe",
-            "Lavt",
-            "For lavt"
-        ],
-        "ignore": [
-            "Ikke aktuelt",
-            ""
-        ]
-    },
-    "Hva synes du om nivået på avsluttende eksamen?": {
-        "order": [
-            "For høyt",
-            "Høyt",
-            "Passe",
-            "Lavt",
-            "For lavt"
-        ],
-        "all": [
-            "Ikke aktuelt",
-            "",
-            "For høyt",
-            "Høyt",
-            "Passe",
-            "Lavt",
-            "For lavt"
-        ],
-        "ignore": [
-            "Ikke aktuelt",
-            ""
-        ]
-    },
-    "How do you rate the course in general?": {
-        "order": [
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "all": [
-            "",
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "ignore": [
-            ""
-        ]
-    },
-    "How do you rate the level of the course?": {
-        "order": [
-            "Too difficult",
-            "Difficult",
-            "OK",
-            "Easy",
-            "Too easy"
-        ],
-        "all": [
-            "",
-            "Too difficult",
-            "Difficult",
-            "OK",
-            "Easy",
-            "Too easy"
-        ],
-        "ignore": [
-            ""
-        ]
-    },
-    "How do you rate the work load in proportion to the number of credits achieved?": {
-        "order": [
-            "Too large",
-            "Large",
-            "OK",
-            "Small",
-            "Too small"
-        ],
-        "all": [
-            "",
-            "Too large",
-            "Large",
-            "OK",
-            "Small",
-            "Too small"
-        ],
-        "ignore": [
-            ""
-        ]
-    },
-    "The quality of the lectures?": {
-        "order": [
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "all": [
-            "No opinion",
-            "",
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "ignore": [
-            "No opinion",
-            ""
-        ]
-    },
-    "The lectures' relevance and usefulness in regard to relaying what you need to learn?": {
-        "order": [
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "all": [
-            "No opinion",
-            "",
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "ignore": [
-            "No opinion",
-            ""
-        ]
-    },
-    "The quality of the group tuition as a whole?": {
-        "order": [
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "all": [
-            "No opinion",
-            "",
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "ignore": [
-            "No opinion",
-            ""
-        ]
-    },
-    "In what degree does group tuition and weekly assignments give you practical skills and the possibility to make use of the theory at hand?": {
-        "order": [
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "all": [
-            "No opinion",
-            "",
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "ignore": [
-            "No opinion",
-            ""
-        ]
-    },
-    "Are the curriculum and learning outcomes of the course sufficiently defined?": {
-        "order": [
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "all": [
-            "No opinion",
-            "",
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "ignore": [
-            "No opinion",
-            ""
-        ]
-    },
-    "How do you rate the course material used (text book etc.)?": {
-        "order": [
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "all": [
-            "No opinion",
-            "",
-            "Exceptionally good",
-            "Very good",
-            "Good",
-            "OK",
-            "Not that good",
-            "Not good"
-        ],
-        "ignore": [
-            "No opinion",
-            ""
-        ]
-    },
-    "How do you rate the difficulty level of the obligatory assignments?": {
-        "order": [
-            "Too high",
-            "High",
-            "OK",
-            "Low",
-            "Too low"
-        ],
-        "all": [
-            "Not relevant",
-            "",
-            "Too high",
-            "High",
-            "OK",
-            "Low",
-            "Too low"
-        ],
-        "ignore": [
-            "Not relevant",
-            ""
-        ]
-    },
-    "What is your opinion in regard to the number of obligatory assignments?": {
-        "order": [
-            "Too high",
-            "High",
-            "OK",
-            "Low",
-            "Too low"
-        ],
-        "all": [
-            "Not relevant",
-            "",
-            "Too high",
-            "High",
-            "OK",
-            "Low",
-            "Too low"
-        ],
-        "ignore": [
-            "Not relevant",
-            ""
-        ]
-    },
-    "How do you rate the difficulty level of the final examination?": {
-        "order": [
-            "Too high",
-            "High",
-            "OK",
-            "Low",
-            "Too low"
-        ],
-        "all": [
-            "Not relevant",
-            "",
-            "Too high",
-            "High",
-            "OK",
-            "Low",
-            "Too low"
-        ],
-        "ignore": [
-            "Not relevant",
-            ""
-        ]
-    }
+  "Hva er ditt generelle intrykk av kurset?": {
+    "order": [
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Greit",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "all": [
+      "",
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Greit",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "ignore": [
+      ""
+    ]
+  },
+  "Hva synes du om emnets nivå?": {
+    "order": [
+      "For vanskelig",
+      "Vanskelig",
+      "Noe vanskelig",
+      "Passe",
+      "Noe lett",
+      "Lett",
+      "For lett"
+    ],
+    "all": [
+      "",
+      "Vanskelig",
+      "For lett",
+      "For vanskelig",
+      "Noe vanskelig",
+      "Passe",
+      "Noe lett",
+      "Lett"
+    ],
+    "ignore": [
+      ""
+    ]
+  },
+  "Hva synes du om arbeidsmengden sett i forhold til antall studiepoeng?": {
+    "order": [
+      "For høy",
+      "For stor",
+      "Høy",
+      "Stor",
+      "Passe",
+      "Lav",
+      "Liten",
+      "For lav",
+      "For liten"
+    ],
+    "all": [
+      "",
+      "Stor",
+      "For stor",
+      "Liten",
+      "For liten",
+      "For høy",
+      "Høy",
+      "Passe",
+      "Lav",
+      "For lav"
+    ],
+    "ignore": [
+      ""
+    ]
+  },
+  "Kvaliteten på forelesningene?": {
+    "order": [
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "all": [
+      "Vet ikke",
+      "",
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "ignore": [
+      "Vet ikke",
+      ""
+    ]
+  },
+  "Forelesningenes relevans og nytteverdi i forhold til å formidle det du trenger å lære?": {
+    "order": [
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "all": [
+      "Vet ikke",
+      "",
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "ignore": [
+      "Vet ikke",
+      ""
+    ]
+  },
+  "Kvaliteten på gruppeundervisningen i sin helhet?": {
+    "order": [
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "all": [
+      "Vet ikke",
+      "",
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "ignore": [
+      "Vet ikke",
+      ""
+    ]
+  },
+  "I hvilken grad forbedrer ukesoppgavene og gruppeundervisningen din forståelse av pensum?": {
+    "order": [
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "all": [
+      "Vet ikke",
+      "",
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "ignore": [
+      "Vet ikke",
+      ""
+    ]
+  },
+  "Er pensum og læringsmål i emnet godt definert?": {
+    "order": [
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "all": [
+      "Vet ikke",
+      "",
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "ignore": [
+      "Vet ikke",
+      ""
+    ]
+  },
+  "Hvordan vurderer du kursmateriellet som benyttes (lærebok m.m.)?": {
+    "order": [
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "all": [
+      "Vet ikke",
+      "",
+      "Særdeles bra",
+      "Meget bra",
+      "Bra",
+      "Grei",
+      "Mindre bra",
+      "Lite bra"
+    ],
+    "ignore": [
+      "Vet ikke",
+      ""
+    ]
+  },
+  "Hvordan er nivået på obligene?": {
+    "order": [
+      "For høyt",
+      "Høyt",
+      "Passe",
+      "Lavt",
+      "For lavt"
+    ],
+    "all": [
+      "Ikke aktuelt",
+      "",
+      "For høyt",
+      "Høyt",
+      "Passe",
+      "Lavt",
+      "For lavt"
+    ],
+    "ignore": [
+      "Ikke aktuelt",
+      ""
+    ]
+  },
+  "Hva synes du om antallet obliger?": {
+    "order": [
+      "For høyt",
+      "Høyt",
+      "Passe",
+      "Lavt",
+      "For lavt"
+    ],
+    "all": [
+      "Ikke aktuelt",
+      "",
+      "For høyt",
+      "Høyt",
+      "Passe",
+      "Lavt",
+      "For lavt"
+    ],
+    "ignore": [
+      "Ikke aktuelt",
+      ""
+    ]
+  },
+  "Hva synes du om nivået på avsluttende eksamen?": {
+    "order": [
+      "For høyt",
+      "Høyt",
+      "Passe",
+      "Lavt",
+      "For lavt"
+    ],
+    "all": [
+      "Ikke aktuelt",
+      "",
+      "For høyt",
+      "Høyt",
+      "Passe",
+      "Lavt",
+      "For lavt"
+    ],
+    "ignore": [
+      "Ikke aktuelt",
+      ""
+    ]
+  },
+  "How do you rate the course in general?": {
+    "order": [
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "all": [
+      "",
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "ignore": [
+      ""
+    ]
+  },
+  "How do you rate the level of the course?": {
+    "order": [
+      "Too difficult",
+      "Difficult",
+      "OK",
+      "Easy",
+      "Too easy"
+    ],
+    "all": [
+      "",
+      "Too difficult",
+      "Difficult",
+      "OK",
+      "Easy",
+      "Too easy"
+    ],
+    "ignore": [
+      ""
+    ]
+  },
+  "How do you rate the work load in proportion to the number of credits achieved?": {
+    "order": [
+      "Too large",
+      "Large",
+      "OK",
+      "Small",
+      "Too small"
+    ],
+    "all": [
+      "",
+      "Too large",
+      "Large",
+      "OK",
+      "Small",
+      "Too small"
+    ],
+    "ignore": [
+      ""
+    ]
+  },
+  "The quality of the lectures?": {
+    "order": [
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "all": [
+      "No opinion",
+      "",
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "ignore": [
+      "No opinion",
+      ""
+    ]
+  },
+  "The lectures' relevance and usefulness in regard to relaying what you need to learn?": {
+    "order": [
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "all": [
+      "No opinion",
+      "",
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "ignore": [
+      "No opinion",
+      ""
+    ]
+  },
+  "The quality of the group tuition as a whole?": {
+    "order": [
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "all": [
+      "No opinion",
+      "",
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "ignore": [
+      "No opinion",
+      ""
+    ]
+  },
+  "In what degree does group tuition and weekly assignments give you practical skills and the possibility to make use of the theory at hand?": {
+    "order": [
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "all": [
+      "No opinion",
+      "",
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "ignore": [
+      "No opinion",
+      ""
+    ]
+  },
+  "Are the curriculum and learning outcomes of the course sufficiently defined?": {
+    "order": [
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "all": [
+      "No opinion",
+      "",
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "ignore": [
+      "No opinion",
+      ""
+    ]
+  },
+  "How do you rate the course material used (text book etc.)?": {
+    "order": [
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "all": [
+      "No opinion",
+      "",
+      "Exceptionally good",
+      "Very good",
+      "Good",
+      "OK",
+      "Not that good",
+      "Not good"
+    ],
+    "ignore": [
+      "No opinion",
+      ""
+    ]
+  },
+  "How do you rate the difficulty level of the obligatory assignments?": {
+    "order": [
+      "Too high",
+      "High",
+      "OK",
+      "Low",
+      "Too low"
+    ],
+    "all": [
+      "Not relevant",
+      "",
+      "Too high",
+      "High",
+      "OK",
+      "Low",
+      "Too low"
+    ],
+    "ignore": [
+      "Not relevant",
+      ""
+    ]
+  },
+  "What is your opinion in regard to the number of obligatory assignments?": {
+    "order": [
+      "Too high",
+      "High",
+      "OK",
+      "Low",
+      "Too low"
+    ],
+    "all": [
+      "Not relevant",
+      "",
+      "Too high",
+      "High",
+      "OK",
+      "Low",
+      "Too low"
+    ],
+    "ignore": [
+      "Not relevant",
+      ""
+    ]
+  },
+  "How do you rate the difficulty level of the final examination?": {
+    "order": [
+      "Too high",
+      "High",
+      "OK",
+      "Low",
+      "Too low"
+    ],
+    "all": [
+      "Not relevant",
+      "",
+      "Too high",
+      "High",
+      "OK",
+      "Low",
+      "Too low"
+    ],
+    "ignore": [
+      "Not relevant",
+      ""
+    ]
+  },
+  "Hva er ditt generelle inntrykk av kurset?": {
+    "order": [
+      "Meget bra",
+      "Ganske bra",
+      "Noe bra",
+      "Hverken bra eller dårlig",
+      "Noe dårlig",
+      "Ganske dårlig",
+      "Meget dårlig"
+    ],
+    "all": [
+      "Ganske bra",
+      "Hverken bra eller dårlig",
+      "Noe bra",
+      "Ganske dårlig",
+      "Noe dårlig",
+      "Meget dårlig",
+      "Meget bra"
+    ],
+    "ignore": []
+  },
+  "What is your general impression of the course?": {
+    "order": [
+      "Exceptionally good",
+      "Very good",
+      "Somewhat good",
+      "Neither good nor bad",
+      "Somewhat bad",
+      "Very bad",
+      "Exceptionally bad"
+    ],
+    "all": [
+      "Neither good nor bad",
+      "Somewhat good",
+      "Somewhat bad",
+      "Very bad",
+      "Exceptionally bad",
+      "Exceptionally good",
+      "Very good"
+    ],
+    "ignore": []
+  },
+  "What do you think of the course's difficulty level?": {
+    "order": [
+      "Too difficult",
+      "Difficult",
+      "Appropriate",
+      "Easy",
+      "Too easy"
+    ],
+    "all": [
+      "Appropriate",
+      "Too difficult",
+      "Difficult",
+      "Easy",
+      "Too easy"
+    ],
+    "ignore": []
+  },
+  "What do you think of the amount of work required in relation to a number of credits given?": {
+    "order": [
+      "Too large",
+      "Large",
+      "Appropriate",
+      "Small",
+      "Too small"
+    ],
+    "all": [
+      "Appropriate",
+      "",
+      "Too large",
+      "Large",
+      "Small",
+      "Too small"
+    ],
+    "ignore": [
+      ""
+    ]
+  },
+  "What is your general impression of the quality of the lectures?": {
+    "order": [
+      "Exceptionally good",
+      "Very good",
+      "Somewhat good",
+      "Neither good nor bad",
+      "Somewhat bad",
+      "Very bad",
+      "Exceptionally bad"
+    ],
+    "all": [
+      "Somewhat good",
+      "",
+      "Somewhat bad",
+      "Neither good nor bad",
+      "Very bad",
+      "Exceptionally bad",
+      "Exceptionally good",
+      "Very good"
+    ],
+    "ignore": [
+      ""
+    ]
+  }
 }

--- a/src/course.py
+++ b/src/course.py
@@ -44,7 +44,7 @@ def generate_stats(responses, participation, scales, stats=None):
                 elif "How" in question or "What" in question:
                     language = "EN"
                 else:
-                    print("Unable to detect language:")
+                    print("Unable to detect language in question:")
                     print(question)
                     sys.exit(1)
             question_answers = responses[question]
@@ -99,8 +99,17 @@ def generate_stats_file(responses_path, participation_path, output_path, scales,
     stats = OrderedDict()
     stats["course"] = course
     stats = generate_stats(responses, participation, scales, stats)
-    if stats is not None:
-        dump_json(stats, output_path)
+    if not stats:
+        print("Skipping course with 0 answers:")
+        print(json.dumps(course, indent=2))
+        return
+    if not stats["language"]:
+        print("Unable to detect language in course:")
+        print(json.dumps(course, indent=2))
+        print("This most likely means that the questions have changed and need to be added to scales.json")
+        sys.exit(1)
+
+    dump_json(stats, output_path)
 
 def generate_stats_dir(responses_dir, participation_dir, output_dir, scales, course_names, semester_name):
     for filename in os.listdir(responses_dir):

--- a/src/pdf.sh
+++ b/src/pdf.sh
@@ -1,3 +1,4 @@
+#!/usr/bin/env bash
 function triplelatex {
     filename=$(basename "$1")
     filename="${filename%.*}"
@@ -15,5 +16,5 @@ function triplelatex {
 }
 
 mkdir -p ./data/$1/outputs/report
-cd ./data/$1/outputs/report
+cd ./data/$1/outputs/report || exit 1
 triplelatex fui-kk_report_$1.tex

--- a/src/semester.py
+++ b/src/semester.py
@@ -19,14 +19,22 @@ def get_course_data(path):
     language = course["language"]
 
     if language == "NO":
-        general_question = "Hva er ditt generelle intrykk av kurset?"
+        general_questions = ["Hva er ditt generelle intrykk av kurset?",
+                             "Hva er ditt generelle inntrykk av kurset?"]
     elif language == "EN":
-        general_question = "How do you rate the course in general?"
+        general_questions = ["How do you rate the course in general?",
+                             "What is your general impression of the course?"]
     else:
-        print("Unknown language: "+language)
+        print("Unknown language: " + str(language))
+        print("Path:" + str(path))
+        print("JSON:")
+        print(json.dumps(course, indent=2))
         sys.exit(1)
 
-    course[general_question] = course["questions"][general_question]
+    for general_question in general_questions:
+        if general_question in course["questions"]:
+            course[general_question] = course["questions"][general_question]
+            break
     del course["questions"]
     return course
 

--- a/src/tex.sh
+++ b/src/tex.sh
@@ -1,3 +1,4 @@
+#!/usr/bin/env/bash
 mkdir -p ./data/$1/outputs/tex
 mkdir -p ./data/$1/inputs/md
 mkdir -p ./data/$1/outputs/report
@@ -19,5 +20,5 @@ if [ ! -f ./data/$1/inputs/tex/tail.tex ]; then
   cp ./templates/tail.tex ./data/$1/inputs/tex/tail.tex
 fi
 
-cd ./data/$1/inputs/md
+cd ./data/$1/inputs/md || exit 1
 find . -iname "*.md" -type f -exec sh -c 'pandoc "${0}" -o "../../outputs/tex/${0%.md}.tex"' {} \;

--- a/src/web.sh
+++ b/src/web.sh
@@ -1,3 +1,4 @@
+#!/usr/bin/env bash
 mkdir -p ./data/$1/outputs/web/converted
 mkdir -p ./data/$1/outputs/web/upload/$1/stats/
 
@@ -6,5 +7,5 @@ cp -r ./resources/d3-charts/dist/ ./data/$1/outputs/web/upload/$1/
 cp -r ./data/$1/outputs/stats/ ./data/$1/outputs/web/upload/$1/stats/
 rename -f "s/\.json$/.js/" ./data/$1/outputs/web/upload/$1/stats/*
 
-cd ./data/$1/inputs/md
+cd ./data/$1/inputs/md || exit 1
 find . -iname "*.md" -type f -exec sh -c 'pandoc "${0}" -o "../../outputs/web/converted/${0%.md}.html"' {} \;

